src/VoteHelper.ts

```diff
@@ -0,0 +1,161 @@
+            }
+        })
+
+        return ongoingReferendums
+    }
+
+    async getAccountVotes(accountAddress: string): Promise<Vote[]> {
+        const accountVotes: Vote[] = []
+        const uniqueVotes = new Set<number>();
+
+        const referendaTracks = this.api.consts.referenda.tracks
+
+        const trackArray = (referendaTracks as any).map(([key, value]) => {
+            return new TrackVotes(key, value)
+        });
+
+        for (const track of trackArray) {
+            const votesInTrack = await this.api.query.convictionVoting.votingFor(accountAddress, track.trackNumber);
+            (votesInTrack as any).asCasting.votes.map((voteData) => {
+                const vote = new Vote(voteData)
+                if (!uniqueVotes.has(vote.referendaNumber)) {
+                    uniqueVotes.add(vote.referendaNumber)
+                    accountVotes.push(vote)
+                }
+            })
+        }
+
+        return accountVotes
+    }
+
+    async getPotentialVotes(network: NetworkConfig, referendums: Referendum[]): Promise<Vote[]> {
+        const uniqueVotes = new Set<number>();
+        const newVoteForReferenda: Vote[] = [];
+        for (const account of network.votersAccounts) {
+            const voterVotes = await this.getAccountVotes(account);
+
+            for (const referenda of referendums) {
+                const vote = voterVotes.find(voterVote => voterVote.referendaNumber === referenda.number);
+                if (vote) {
+                    if (!uniqueVotes.has(vote.referendaNumber)) {
+                        uniqueVotes.add(vote.referendaNumber)
+                        newVoteForReferenda.push(vote)
+                    }
+                }
+            }
+        }
+
+        return newVoteForReferenda
+    }
+
+    async decideHowToVote(validator: string, potentialVotes: Vote[]): Promise<Vote[]> {
+
+        function compareVotes(vote1: Vote, vote2: Vote): boolean {
+            return vote1.direction == vote2.direction && vote1.referendaNumber == vote2.referendaNumber;
+        }
+
+        const validatorVotes = await this.getAccountVotes(validator);
+
+        for (let i = potentialVotes.length - 1; i >= 0; i--) {
+            const vote = potentialVotes[i];
+            // Check if the direction parameter of vote and any of the validatorVotes are the same
+            if (validatorVotes.some(v => compareVotes(vote, v))) {
+                potentialVotes.splice(i, 1); // Remove the vote from potentialVotes
+            }
+        }
+
+        return potentialVotes
+    }
+
+    async voteForReferenda(vote: Vote, sender, validator, balance): Promise<void> {
+        const constructedVote = this.api.createType('PalletConvictionVotingVoteAccountVote', {
+            'Standard': {
+                'vote': {
+                    'aye': vote.direction,
+                    'conviction': 'Locked1x',
+                },
+                'balance': balance
+            }
+        })
```

Comment:
Here we create vote call, direction is taking from voters direction, balance is hardcoded in config file